### ISSUE TITLE
bcftools fails with no error message on AWS

### ISSUE THREAD

**Issue by bhandsaker** [samtools/bcftools#1065]
I am trying to run bcftools 1.9 on Amazon AWS. I am running the following command (reading from an EBS drive and writing to the same EBS drive):

```
bcftools view -O u -f .,PASS -c 5 /mnt/bcfs/large_input_chr1.gz \
    | bcftools annotate -O b -x '^FORMAT/GT,FORMAT/GQ' \
    > /mnt/bcfs/chr1_AC5.bcf
```
On a smaller input file (chr22, about 15G) I have gotten this to run successfully, but when I try to run it on chr1 (500G) it fails with no error message, leaving a truncated output bcf. It has failed twice now, in one case getting about 5% of the way through, another time about 15%.

Do you have any suggestions about how to debug this? Should bcftools produce some kind of message if it gets an I/O error? My understanding is that Amazon EBS drives are fairly reliable (similar to NFS), which is why I'm staging the data there rather than trying to access S3 storage directly.

**Comment by pd3** [comment]
Could this be related to https://github.com/samtools/bcftools/issues/1059?

BCFtools checks the return status of all write calls and should throw an error if not successful. One way to debug this would be to modify vcfview.c and vcfannotate.c and make them fprintf to stderr the chr+position and see what gets dropped. Also saving the output of the first command into an intermediate file before feeding it to the other will help to isolate the problem.

**Comment by biozzq** [comment]
Dear @pd3 

I found that many commands related to the filtering process do not have screen output. It will be better for us to know the number of excluded sites for each different filter strategy, such as how many SNPs has been excluded by the missing rate.

Best wishes,
Zhuqing

**Comment by pd3** [comment]
@biozzq Yeah, it would be useful to add that. Probably best to open a separate issue for that and hope it will be picked on at some point.

@bhandsaker Were you able to narrow the problem down / is it reproducible? If you are using bash, do you have the pipefail option set, `set -o pipefail on`?

**Comment by bhandsaker** [comment]
@pd3 Yes, I was able to resolve the AWS issue, which it turned out had nothing to do with bcftools per se. But I'll record it here for posterity.

The problem was that I was running processes in the background that were not completely detached from a pty. What confused us was that on AWS apparently the ptys can be held open for many hours by sshd after the other end has been closed (i.e. ssh going away). Once we realized the failures (process terminations) coincided with the ptys being torn down, it all made sense.

Thanks for your help!
